src/mesh/NodeDB.cpp

```diff
@@ -710,9 +710,8 @@ void NodeDB::updateUser(uint32_t nodeId, const meshtastic_User &p)
         powerFSM.trigger(EVENT_NODEDB_UPDATED);
         notifyObservers(true); // Force an update whether or not our node counts have changed
 
-        // Not really needed - we will save anyways when we go to sleep
         // We just changed something important about the user, store our DB
-        // saveToDisk();
+        saveToDisk(SEGMENT_DEVICESTATE);
```

Comment:
The old call was in doDeepSleep (sleep.cpp 204) and was saving everything. Is devicestate enough here?

Reply:
I guessed so, because this function is triggered (only) when a node changes. And the DEVICESTATE segment contains the node info. Other segments are handled during configurations already where we don't have issues.

Reply:
I did some more digging into this and noticed another (albeit minor) side effect:
MeshService::reloadOwner calls this NodeDB::updateUser method and it has it's own saveToDisk(SEGMENT_DEVICESTATE) call. This would result in an unnecessary write to disk under certain circumstances

Reply:
> I did some more digging into this and noticed another (albeit minor) side effect: MeshService::reloadOwner calls this NodeDB::updateUser method and it has it's own saveToDisk(SEGMENT_DEVICESTATE) call. This would result in an unnecessary write to disk under certain circumstances

In turn that would mean removing the call in reloadOwner. Since every call we make to reloadOwner saves devicestate immediately afterwards anyway.

Reply:
ref: AdminModule.cpp 234 and 612,

Reply:
> > I did some more digging into this and noticed another (albeit minor) side effect: MeshService::reloadOwner calls this NodeDB::updateUser method and it has it's own saveToDisk(SEGMENT_DEVICESTATE) call. This would result in an unnecessary write to disk under certain circumstances
> 
> In turn that would mean removing the call in reloadOwner. Since every call we make to reloadOwner saves devicestate immediately afterwards anyway.

yes, that makes sense... we don't need three saveToDisk when the owner name changes;-)